framework/src/bundle/BundlePrivate.cpp

```diff
@@ -582,73 +582,77 @@ void BundlePrivate::Start(uint32_t options)
 
 std::exception_ptr BundlePrivate::Start0()
 {
-  bool bStarted = false;
   // res is used to signal that start did not complete in a normal way
   std::exception_ptr res;
-
-  coreCtx->listeners.BundleChanged(BundleEvent(BundleEvent::BUNDLE_STARTING, MakeBundle(this->shared_from_this())));
-
-  try
+  auto const thisBundle = MakeBundle(this->shared_from_this());
+  coreCtx->listeners.BundleChanged(BundleEvent(BundleEvent::BUNDLE_STARTING, thisBundle));
+  
+  Any bundleActivatorVal = thisBundle.GetProperty(Constants::BUNDLE_ACTIVATOR);
+  bool useActivator = bundleActivatorVal.Empty() ? false : any_cast<bool>(bundleActivatorVal);
```

Comment:
This will throw if `useActivator` is not of type `bool`. Probably not what we want.